pandas/tests/groupby/transform/test_transform.py

```diff
@@ -1289,3 +1289,13 @@ def test_transform_cumcount():
 
     result = grp.transform("cumcount")
     tm.assert_series_equal(result, expected)
+
+
+def test_group_mean_timedelta_nat():
```

Comment:
can you repeat this test with dt64 and dt64tz

also with min_count etc

Reply:
Thanks, wasn't thinking about different timeseries datatypes. Fully agree on more tests and I had added an additional one but removed it in the hope of getting CI to pass.

 I'll add the following:
 * min_count= [0, -1]
 * input_values datatype = {dt64,dt64tz}
 * datetimelike=false but datetime array
 * datetimelike=true

Reply:
Done.